packages/shared/src/components/modals/common/Modal.tsx

```diff
@@ -13,6 +13,43 @@ export type ModalProps = ReactModal.Props & {
   size?: ModalSize;
 };
 
+const modalKindToOverlayClassName: Record<ModalKind, string> = {
+  [ModalKind.FixedCenter]: 'mobileL:justify-center',
+  [ModalKind.FlexibleCenter]: 'justify-center',
+  [ModalKind.FlexibleTop]: 'm-0',
+};
+const modalKindAndSizeToOverlayClassName: Partial<
+  Record<ModalKind, Partial<Record<ModalSize, string>>>
+> = {
+  [ModalKind.FlexibleTop]: {
+    [ModalSize.Medium]: 'mobileL:pt-10',
+  },
+};
+const modalKindToClassName: Record<ModalKind, string> = {
+  [ModalKind.FixedCenter]:
+    'h-full max-h-[calc(100vh-2.5rem)] mobileL:h-[40rem] mobileL:max-h-[calc(100vh-5rem)] mt-10 mobileL:mt-0',
+  [ModalKind.FlexibleCenter]:
+    'mx-4 max-w-[calc(100vw-2rem)] max-h-[min(calc(100vh),40rem)] mobileL:max-h-[min(calc(100vh-5rem),40rem)]',
```

Comment:
Nothing wrong with this, but when using vh and calc it's very important to solidly test it out on actual mobile devices.
I know safari on iPhone has some issues with vh.

Reply:
Yeap, makes perfect sense, I left the mobile device tests out of this. Let me add a jira just for this :
https://dailydotdev.atlassian.net/browse/WT-734